Added ML with Python

r0f1 opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->


<!-- Please fill in the below placeholders -->

https://github.com/r0f1/datascience

This list is a personal collection of datascience libraries that I find helpful. I code mainly in Python, so almost everything on the list is Python-related. I tried to also link Youtube Videos, IPython Notebook files, and additional information, wherever I thought it was a good fit. 

I think this is a good addition to the awesome list, because the other datascience list, has not been updated in years, and no other list here, contains the information I have.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.


## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

angrykoala commented:
Hi @r0f1 
A few improvements I would recommend on this PR:

* Move the item [ML wit Python] to the bottom of the section (right under H2O)
* Your list title ("Data Science") doesn't indicate that this is for Python
* You list doesn't comply with the awesome style formatting `- [AVA](…) - JavaScript test runner.`
* If possible, I would add a TOC

r0f1 commented:
Hi @angrykoala, thanks for getting back to me so quickly. I implemented all of your suggested changes. I moved the item below H2O, changed my title ("Data Science with Python") to more appropriately reflect its content, changed all pipes to dashes and added a TOC.

catalinmiron commented:
Thanks for contributing @r0f1 

Take a quick look:
- The heading should be Awesome Data Science with Python, not Data Science with Python.
- I would suggest you create a LICENSE file to officially make a CC0 repository. Take a look at this repository as an example (https://github.com/victorshinya/awesome-ibmcloud/blob/master/LICENSE).
- Includes a succinct description of the project/theme at the top of the readme.
- The `Table of Contents` should be named Contents, not Table of Contents or TOC.
- Please create a contributing guide

r0f1 commented:
Hi @catalinmiron, I implemented all of the changes you suggested but chose a different licence than the one you proposed. I clicked through a couple of other awesome lists, and saw that authors use a variety of licences. I chose the MIT licence.

sindresorhus commented:
> I chose the MIT licence.

🤦‍♂️ ⬇️ 

> Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
> * That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**

sindresorhus commented on readme.md:
```diff
@@ -284,6 +284,7 @@
 	- [ML with Ruby](https://github.com/arbox/machine-learning-with-ruby#readme) - Learning, implementing, and applying Machine Learning using Ruby.
 	- [Core ML Models](https://github.com/likedan/Awesome-CoreML-Models#readme) - Models for Apple's machine learning framework.
 	- [H2O](https://github.com/h2oai/awesome-h2o#readme) - Open source distributed machine learning platform written in Java with APIs in R, Python, and Scala.
+	- [ML with Python](https://github.com/r0f1/datascience#readme)
```
Needs a description.

sindresorhus replied:
`ML with Python` is also not the title of the list.

sindresorhus commented:
```
❯ awesome-lint https://github.com/r0f1/datascience
✖ Linting

  README.md:51:126
  ✖    1:1    Missing or invalid Table of Contents                                                               remark-lint:awesome/toc
  ✖    6:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖    7:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖    8:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖    9:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   10:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   11:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   12:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   13:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   14:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   15:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   16:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   17:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   18:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   19:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   20:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   21:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   22:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   23:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   24:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   25:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   26:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   27:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   28:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   29:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   30:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   31:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   32:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   33:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   34:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   35:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   36:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   37:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   38:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   39:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   40:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   41:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   42:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   43:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   44:3    Invalid list item link URL                                                                         remark-lint:awesome/list-item
  ✖   51:126  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   53:108  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   54:73   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   59:131  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   60:74   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   61:105  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   65:75   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   66:44   Text "Github" should be written as "GitHub"                                                        remark-lint:awesome/spell-check
  ✖   74:158  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   75:85   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   76:125  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   77:60   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   81:85   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖   99:82   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  101:229  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  117:127  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  123:347  Link to https://github.com/WillKoehrsen/Data-Analysis/blob/master/medium/2019-01-13_stats is dead  remark-lint:no-dead-urls
  ✖  123:437  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  127:83   Link to https://demo.bokehplots.com/ is dead                                                       remark-lint:no-dead-urls
  ✖  127:273  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  131:68   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  134:89   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  135:249  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  136:77   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  146:153  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  147:80   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  148:52   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  150:23   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  151:30   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  155:68   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  168:59   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  169:122  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  180:91   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  181:350  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  182:178  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  184:364  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  186:34   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  187:87   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  188:74   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  193:193  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  194:87   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  195:121  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  196:78   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  205:107  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  212:91   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  213:104  Link to https://www.kaggle.com/ashishpatel26/genetic-algorithm-for-beginner is dead                remark-lint:no-dead-urls
  ✖  214:99   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  218:305  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  219:81   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  225:82   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  226:92   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  230:82   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  231:84   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  235:98   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  236:118  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  237:98   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  238:86   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  239:158  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  240:107  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  241:73   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  242:91   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  243:79   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  244:96   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  274:90   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  275:102  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  276:111  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  279:134  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  280:111  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  281:81   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  282:113  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  285:309  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  288:144  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  289:141  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  296:82   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  298:234  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  309:80   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  311:80   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  319:97   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  327:194  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  345:281  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  365:87   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  368:72   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  373:2    Text "Youtube" should be written as "YouTube"                                                      remark-lint:awesome/spell-check
  ✖  373:87   Text "Youtube" should be written as "YouTube"                                                      remark-lint:awesome/spell-check
  ✖  373:169  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  374:254  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  375:321  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  385:130  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  390:70   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  391:94   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  399:79   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  405:80   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  406:84   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  407:92   Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  411:111  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces
  ✖  412:101  Use two spaces for hard line breaks                                                                remark-lint:hard-break-spaces

  136 errors
```

https://github.com/sindresorhus/awesome-lint

sindresorhus commented:
> * You list doesn't comply with the awesome style formatting `- [AVA](…) - JavaScript test runner.`

This is not fixed.

sindresorhus commented:
> * **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)

r0f1 commented:
Hi, I fixed some, but not all issues.
Fixed: I added a short description to the list, changed the license, and added the succinct description. 
However: I did not fix all linter errors. It complains that my TOC is not correct (this is most likely due to the fact that I do not display all headings) and that my headings are not correct (I use almost always four # and neglect a proper hierarchy). I guess that the latter issue is a cause for the former one. Anyways, I don't really want to fix these issues, because not all headings are necessary in the TOC and the default heading hierarchy would produce very large headings with padding around them which impedes readability. 

 The style `* [AVA](…) - JavaScript test runner` is adhered to in most cases. I don't see why violating this rule would be much on an issue.

sindresorhus commented:
Sorry, I don't have energy to argue. Just gonna pass on this as it's very far from being acceptable for inclusion here.
